### PR TITLE
Enable gziping

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

Enables gziping in for pycherry, django, and webpack server (just for fun).
Increases Lighthouse performance score of by 20 points.
To test ` yarn run build && kolibri start`

#### Before

before

#### After

after

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/utils/server.py [hunk 1]
```diff
@@ -133,7 +133,11 @@ def run_server(port):
     from kolibri.deployment.default.wsgi import application
     cherrypy.tree.graft(application, "/")
 
-    cherrypy.config.update({"environment": "production"})
+    cherrypy.config.update({
+        "environment": "production",
+        "tools.gzip.on": True,
+        "tools.gzip.mime_types": ["text/*", "application/javascript"],
+        })
```
Comment (jonboiser) [inline kolibri/utils/server.py]: might need to move this `})` four spaces to the left

**Comment by rtibbles** [comment]
Seems to make a big difference to first meaningful paint. There was a reason that we had not used this in KA Lite, which I think was to do with the C dependencies of zip compression in Python. @lyw07 @benjaoming do you see any issues with this in Kolibri?

**Comment by rtibbles** [comment]
Another way to check issues for this would be to see whether this works on the Android build.

**Comment by lyw07** [comment]
@rtibbles Sorry I'm not really familiar with this... I ran the pex file on my laptop with python 2/3, and it's working well. I also checked the source code of `GZipMiddleware`, and it seems to be pure python, so I'm not able to find the "C dependencies" included here. Maybe I'm misunderstanding things here? If you have time, could you explain a bit more? Thank you!

**Comment by rtibbles** [comment]
@lyw07 yeah my memory is a little hazy on exactly why we didn't do this in KA Lite previously, am hoping that @benjaoming can recall.

**Comment by benjaoming** [comment]
@rtibbles we avoid zip compression in KA Lite for zip files distributing the content packs. This was either because of processor usage or the compression algorithm being unavailable... I don't know which. As for the middleware, I have no idea.

But again, consider that there will be a processor doing work to compress things on demand. This does not scale, so should be made configurable.

@christianmemije should both Django *and* cherrypy be having this enabled?

**Comment by christianmemije** [comment]
So I played around with creating gzip assets as part of the webpack build process, but I ran into [an issue](https://github.com/cherrypy/cherrypy/issues/1558) with CherryPy. Well the GZipMiddleware is for the data served by Django, including the html file and other responses. And the gzip compression by CherryPy is for all static css and js files. So the compression by CherryPy would only occur on first request, and after that the browser would ideally just be referencing the cached assets.

**Comment by benjaoming** [comment]
Ah yes, @christianmemije - thanks for the reminder, that's so true that the cherry-py does other things.. might it not also just cache the responses from Django, since it also serves those?

> So the compression by CherryPy would only occur on first request, and after that the browser would ideally just be referencing the cached assets.

This is client-side, though: What I mean is if everything that is requested has to be cached, then we would have +x% load on the server-side for each client added. It seems it might be out-weighed by other gains :)

**Pull request merged**